wp-includes/class-wp-service-workers.php

```diff
@@ -51,10 +93,34 @@ class WP_Service_Workers extends WP_Scripts {
 	 */
 	public $output = '';
 
+	/**
+	 * Registered caching routes and scripts.
+	 *
+	 * @var array
+	 */
+	public $registered_caching_routes = array();
+
 	/**
 	 * Initialize the class.
 	 */
 	public function init() {
+
+		if ( ! function_exists( 'list_files' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		$this->register(
+			'workbox-sw',
+			array( $this, 'get_workbox_script' ),
+			array()
+		);
+
+		if ( SCRIPT_DEBUG || is_preview() ) {
```

Comment:
Why `is_preview()`? How will this ever return `true` given that it is the service worker response?

Reply:
Right💡 Will remove.

Reply:
Hmm, actually, at this moment `init` is called on any page if any route caching is registered, not only when serving the request of the registered service worker.